st/clitests/awss3api_spec.py

```diff
@@ -110,6 +112,119 @@ def create_object_list_file(file_name, obj_list=[], quiet_mode="false"):
+    file_to_create = os.path.join(upload_dir, root_file)
+    key = root_file
+    with open(file_to_create, 'wb+') as fout:
+        fout.write(os.urandom(filesize[start]))
+    start = start + 1
+
+#calculating md5sum in dict.
+md5sum_dict_before_upload = {}
+for file in file_list_before_upload:
+    path = os.path.join(upload_dir, file)
+    md5sum_dict_before_upload[file] = get_md5_sum(path)
+
+#Recursively upload all file from upload_dir to 'seagatebucket'.
+AwsTest('Aws Upload folder recursively').upload_objects("copyobjectbucket", upload_dir)\
+        .execute_test().command_is_successful()
+
+# Create a new destination bucket.
+AwsTest('Aws can create destination bucket for CopyObject API')\
+    .create_bucket("destination-bucket")\
+    .execute_test().command_is_successful()
+
+#Copy objects to destination bucket.
+AwsTest('Aws can copy object to different bucket')\
+    .copy_object("copyobjectbucket/1kfile", "destination-bucket", "1kfile")\
+    .execute_test().command_is_successful()
+
+AwsTest('Aws can copy object to different bucket')\
+    .copy_object("copyobjectbucket/5kfile", "destination-bucket", "5kfile")\
+    .execute_test().command_is_successful()
+
+AwsTest('Aws can copy object to different bucket')\
+    .copy_object("copyobjectbucket/32kfile", "destination-bucket", "32kfile")\
+    .execute_test().command_is_successful()
+
+#creating download_dir to download objects.
+download_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "copy_object_test_download")
+os.makedirs(download_dir, exist_ok=True)
+
+#Download keys from destination bucket to local download_dir.
+AwsTest('Aws can Download folder recursively').download_objects("destination-bucket", download_dir)\
+        .execute_test().command_is_successful()
+
+#getting files from download_dir.
+file_list_after_download = os.listdir(download_dir)
+
+#calculating md5sum after downloading the objects.
+md5sum_dict_after_download = {}
+for file in file_list_after_download:
+    path = os.path.join(download_dir, file)
+    md5sum_dict_after_download[file] = get_md5_sum(path)
+
+print("***************Checksum test start****************")
+corruption = False
+for file in md5sum_dict_before_upload:
+    if md5sum_dict_before_upload[file] != md5sum_dict_after_download[file]:
+        corruption = True
+        break
+if corruption:
+    print("***************Checksum test Failed****************")
+else:
+    print("***************Checksum test completed ****************")
+
+#deleting objects from source bucket.
+AwsTest('Aws can delete 1kfile from source bucket').delete_object("copyobjectbucket", "1kfile")\
+    .execute_test().command_is_successful()
+
+AwsTest('Aws can delete 5kfile from source bucket').delete_object("copyobjectbucket", "5kfile")\
+    .execute_test().command_is_successful()
+
+AwsTest('Aws can delete 32kfile from source bucket').delete_object("copyobjectbucket", "32kfile")\
+    .execute_test().command_is_successful()
```

Comment:
src and dst objects can be deleted in a loop

Reply:
done